libraries/AP_TemperatureSensor/AP_TemperatureSensor_MLX90614.cpp

```diff
@@ -0,0 +1,65 @@
+#include "AP_TemperatureSensor_MLX90614.h"
+
+#if AP_TEMPERATURE_SENSOR_MLX90614_ENABLED
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_HAL/I2CDevice.h>
+#include <AP_Math/AP_Math.h>
+
+
+extern const AP_HAL::HAL &hal;
+
+#define MLX90614_I2CDEFAULTADDR 0x5A    // Device default slave address 
+#define MLX90614_BROADCASTADDR  0       // Device broadcast slave address 
+
+//  RAM addresses
+#define MLX90614_RAWIR1         0x04    //  RAM reg - Raw temperature, source #1 
+#define MLX90614_RAWIR2         0x05    //  RAM reg - Raw temperature, source #2 
+#define MLX90614_TA             0x06    //  RAM reg - Linearized temperature, ambient 
+#define MLX90614_TOBJ1          0x07    //  RAM reg - Linearized temperature, source #1 
+#define MLX90614_TOBJ2          0x08    //  RAM reg - Linearized temperature, source #2 
+
+void AP_TemperatureSensor_MLX90614::init() {
+
+    if ((_params.bus_address != MLX90614_I2CDEFAULTADDR) ) {
+        _params.bus_address.set(MLX90614_I2CDEFAULTADDR);
+    }
```

Comment:
If its possible to change the address then you might just want to do a set default here so the user can still change the param if they need to.

Reply:
So Should I remove default check which will set address to what user has put in param.

Reply:
Is it possible to change the address on the device? Or buy a version with a different address? If this address is always correct then just setting the param is fine.  Although there is no need to check the value fist, might as well just set it every time.

Reply:
Is there is a way to change the default address of the Sensor by writing into EEPROM of the sensor. After that it needs hard reboot